### PR TITLE
Code fix for SA1213

### LINKED ISSUE #495: Code fix: SA1212 (PropertyAccessorsMustFollowOrder)
Implement a code fix for #72.

### LINKED ISSUE #1105: Code fix: SA1213 (EventAccessorsMustFollowOrder)
Implement a code fix for #73.

### PR DESCRIPTION
**Pull request by hvanbakel**
Fixes #1105 
Fixes #495

### CODE REVIEW

**Comment by hvanbakel** [comment]
Now also a fix for #495

File: StyleCop.Analyzers/StyleCop.Analyzers.Test/OrderingRules/SA1212UnitTests.cs [hunk 1]
```diff
@@ -35,6 +37,27 @@ public int Prop
...
+{
+    private int i = 0;
+
+    public int Prop
+    {
+        get
+        {
+            return i;
+        }
+        set
```
Comment (sharwell) [inline StyleCop.Analyzers/StyleCop.Analyzers.Test/OrderingRules/SA1212UnitTests.cs]: :exclamation: This needs a blank line above it.

File: StyleCop.Analyzers/StyleCop.Analyzers.Test/OrderingRules/SA1212UnitTests.cs [hunk 2]
```diff
@@ -144,6 +180,27 @@ public class Foo
...
+{
+    private int field;
+
+    public int this[int index]
+    {
+        get
+        {
+            return field;
+        }
+        set
```
Comment (sharwell) [inline StyleCop.Analyzers/StyleCop.Analyzers.Test/OrderingRules/SA1212UnitTests.cs]: :exclamation: This needs a blank line above it.

**Comment by sharwell** [comment]
:bulb: This is missing tests for cases where trivia fall in the reordering. For example:

```csharp
public int Property
{
  // Something
  set
  {
  }

  // Something else
  get
  {
    return 0;
  }
}
```

**Comment by hvanbakel** [comment]
Almost done, failing test is due to #1327

**Comment by hvanbakel** [comment]
No longer blocked by #1327, worked around by using a local function

**Pull request merged**